### PR TITLE
Support ImageSet in fit

### PR DESCRIPTION
**Pull request by hkvision**
Fix https://github.com/intel-analytics/zoo/issues/270

cc @dding3 @jenniew @jason-dai 

fit accepts ImageSet as data.
Also adds a missing unit test from 0-based-label PR.

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala [hunk 1]
```diff
@@ -250,6 +257,21 @@ abstract class KerasNet[T: ClassTag](implicit ev: TensorNumeric[T])
...
+    this.fit(toDataSet(x, batchSize), nbEpoch, toDataSet(validationData, batchSize))
+  }
+
+  def fit(
+      x: ImageSet,
+      batchSize: Int,
+      nbEpoch: Int)(implicit ev: TensorNumeric[T]): Unit = {
+    this.fit(toDataSet(x, batchSize), nbEpoch, null)
+  }
+
```
[the remark this answers is not in the document]
Comment (hkvision) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala]: Fit DataSet already exists https://github.com/intel-analytics/analytics-zoo/blob/master/zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala#L192
Am I misunderstanding your meaning? Do you want to do transformer within fit?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala [hunk 2]
```diff
@@ -181,6 +183,11 @@ abstract class KerasNet[T: ClassTag](implicit ev: TensorNumeric[T])
     else null
   }
 
+  private def toDataSet(x: ImageSet, batchSize: Int): DataSet[MiniBatch[T]] = {
+    if (x != null) DataSet.imageFrame(x.toImageFrame()) -> ImageFeatureToMiniBatch[T](batchSize)
+    else null
+  }
```
Comment (dding3) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala]: Can we wrapper ImageSet into DataSet[ImageFeature] just like what DataSet.imageFrame did? With this, we don't need convert between imageFrame and imageSet.

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (jenniew) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala]: I mean wrap ImageSet into DataSet just as Ding said

[the remark this answers is not in the document]
Comment (hkvision) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/models/Topology.scala]: updated.

**Comment by hkvision** [comment]
This PR is finished @jenniew @dding3 @jason-dai

**Comment by hkvision** [comment]
jenkins http://10.239.47.211:18888/job/ZOO-PR-Validation/749/

File: pyzoo/zoo/feature/image/imagePreprocessing.py [hunk 3]
```diff
@@ -30,6 +30,17 @@ def __init__(self, bigdl_type="float", *args):
...
 
 
+class ImageBytesToMat(ImagePreprocessing):
+    """
+    Transform byte array(original image file in byte) to OpenCVMat
+    :param byte_key key that maps byte array
+    :param image_codec specifying the color type of a loaded image, same as in OpenCV.imread.
+     By default is Imgcodecs.CV_LOAD_IMAGE_UNCHANGED
+    """
+    def __init__(self, byte_key="bytes", image_codec=-1, bigdl_type="float"):
```
Comment (hkvision) [inline pyzoo/zoo/feature/image/imagePreprocessing.py]: @dding3 Any comment for what image_codec mapped to python?

**Comment by hkvision** [comment]
Wondering do we need predict and evaluate on ImageSet? If so, any suggestion how to do this?
Currently, AbstractModule has the following:
* predictImage this takes different arguments as ordinary model predict
* evaluateImage this is almost the same as ordinary model evaluate. We just need to change signature from ImageFrame to ImageSet.
@dding3 @jenniew @jason-dai

File: pyzoo/zoo/feature/image/imagePreprocessing.py (hunk 3 above)
Comment (dding3) [reply pyzoo/zoo/feature/image/imagePreprocessing.py]: I used Int, since opencv python is also using Int

**Comment by dding3** [comment]
If other input(A Numpy array or RDD of Sample) supports predict and evaluate, I think it would be nice if ImageSet also supports that. 

In ImageModel, we defined a new `predictImageSet` to predict ImageSet, maybe you can refer that.

**Comment by dding3** [comment]
LGTM

**Comment by hkvision** [comment]
Merge fit ImageSet first. I will add predict and evaluate ImageSet in another PR after further discussion.
If you have more comments, I will also address them in another PR. @jenniew 
Thanks for your review! @dding3

**Pull request merged**